### ISSUE TITLE
Build PRs from forked repositories which are owned by collaborators

### ISSUE THREAD

**Issue by RobertLucian** [travis-ci/travis-ci#8591]
There is a GitHub organization which, for instance, has a repository.

Now, let's say there is a team which works for that organization and they never directly push to the repository and instead, they have their own forks. When they finish a `feature` or a `hotfix`, they make a PR from their branch (of their own fork) to the upstream repository (the one owned by the organization).

Obviously, the team would want to have their PRs (from their forked repositories) built, tested, deployed (etc) and not those that come from external contributors, who don't have any push/pull rights. 
Is this possible and if so, how can it be done?

The team especially wants to be able to build and deploy development packages (think of `packagecloud`) on each PR that's sent to the organization's repository, without having to worry of external contributors who might want to "hijack" the organization's credentials. 

Now, even more than this, would it be possible to also trigger `travis-ci` builds on the collaborators' forked repositories? I suspect each forked repo would require a child ID of the parent repo or something like this.

**I'm asking all these because the team is said to be orientated on working on each of their own forks and when something is done, they PR to the main repo.**

**Comment by BanzaiMan** [comment]
Is there a reason that the teams can't enable Travis CI on their forks?

**Comment by RobertLucian** [comment]
No, that would not pose a problem - the team could enable Travis CI on their forks.

Assuming everyone that's a part of the team enables Travis CI, how would work the process of creating PRs (where changes are PRed into the main repository)?

Last but not least, if the team works on a private repo and each of the team members has its own fork, will all the members have to pay for their own plans? Like if there are 10 members, will they all have to pay a cumulative *10 x how_much_it_costs_going_private* on Travis CI?
